src/hooks/asana.tsx

```diff
@@ -0,0 +1,157 @@
+
+import createPersistedState from 'use-persisted-state';
+import { createStateContext, usePrevious } from 'react-use';
+import { ITask } from "../components/Task";
+import { map } from "lodash";
+import { createClient } from "../lib/asana";
+import { useEffect, useState, useCallback } from "react";
+
+export interface IUseAsanaTasksResult {
+  loading: boolean
+  sync: () => Promise<void>
+  workspace?: string
+  error?: Error
+  tasks: ITask[]
+}
+
+export interface IUseCreateAsanaTaskResult {
+  loading: boolean
+  createTask: (task: ITask) => Promise<void>
+  workspace?: string
+  error?: Error
+  tasks: ITask[]
+}
+
+const [useTasks, AsanaTasksProvider] = createStateContext<ITask[]>([]);
+
+const mapTaskToAsanaTask = ({ description, complete }: ITask) => ({
+  name: description,
+  completed: complete,
+})
+
+const mapAsanaTaskToTask = ({ name, completed }: any): ITask => ({
+  description: name,
+  complete: completed
+})
+
+export const useAsanaTasks = () => {
+  const [credentials] = useAsanaCredentials();
+  const prevCredentials = usePrevious(credentials);
+  const [tasks, setTasks] = useTasks()
+  const [state, setState] = useState<Omit<IUseAsanaTasksResult, 'tasks'>>({
+    sync: async () => {
+      const token = credentials.access_token
+      if (token) {
+        setState({
+          ...state,
+          loading: true
+        })
+        try {
+          const client = createClient(token, credentials.refresh_token)
+          const user = await client.users.me()
+
+          const userId = user.gid;
+          // The user's "default" workspace is the first one in the list, though
```

Comment:
Nice comment.

Reply:
Ha! I was actually going to say the same thing but decided it was going to be weird to call out a "good comment". But now that we're on it, let me pile on :)